docs/pages/components/split-panel.md

```diff
@@ -239,16 +241,102 @@ To snap panels at specific positions while dragging, add the `snap` attribute wi
     transform: translateX(-3px);
   }
 
-  .split-panel-snapping-dots::before {
+  .split-panel-snapping .split-panel-snapping-dots::before {
     left: 100px;
   }
 
-  .split-panel-snapping-dots::after {
+  .split-panel-snapping .split-panel-snapping-dots::after {
     left: 50%;
   }
 </style>
 ```
 
+Or, if you want to snap the panel to every `100px` interval, as well as at 50% of the panel's size, you can use `snap="repeat(100px) 50%"`.
+
+```html:preview
+<div class="split-panel-snapping-repeat">
+  <sl-split-panel snap="repeat(16px) 50%">
+    <div
+      slot="start"
+      style="height: 150px; background: var(--sl-color-neutral-50); display: flex; align-items: center; justify-content: center; overflow: hidden;"
+    >
+      Start
+    </div>
+    <div
+      slot="end"
+      style="height: 150px; background: var(--sl-color-neutral-50); display: flex; align-items: center; justify-content: center; overflow: hidden;"
+    >
+      End
+    </div>
+  </sl-split-panel>
+</div>
+
+<style>
+  .split-panel-snapping-repeat {
+    position: relative;
+  }
+</style>
+```
+
+You can also implement a custom snap function which controls the snapping manually. To do this, you need to acquire a reference to the element in Javascript and set the `snap` property. For example, if you want to snap the divider to either `100px` from the left or `100px` from the right, you can set the `snap` property to a function encoding that logic.
+
+```panel.snap = ({ pos, size }) => (pos < size / 2) ? 100 : (size - 100)```
```

Comment:
```suggestion
```js
panel.snap = ({ pos, size }) => (pos < size / 2) ? 100 : (size - 100)
```
```

Reply:
The suggestion escaping might be broken here. The point was to put it into a proper block with JS.